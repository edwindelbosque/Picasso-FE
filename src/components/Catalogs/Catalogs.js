import React from 'react';
import './Catalogs.scss';
import { NavLink } from 'react-router-dom';
// import { getPalettes } from '../../util/apiCalls';
import Fade from 'react-reveal/Fade';

const Catalogs = ({ catalogs, menuIsActive, updateCurrentCatalog }) => {
	let allCatalogs;

	if (catalogs) {
		allCatalogs = catalogs.map(catalog => {
			const { catalogName, id } = catalog;
			return (
				<NavLink
					onClick={() => updateCurrentCatalog(id)}
					key={id}
					exact
					to={`/catalogs/${id}`}
					activeClassName='active-catalog'>
					{' '}
					<li>{catalogName}</li>
				</NavLink>
			);
		});
	}

	// const fetchPalettes = () => {
	// 	catalogs.forEach(async catalog => {
	// 		const palettess = await getPalettes(catalog);
	// 		console.log(palettess);
	// 	});
	// };

	return (
		<section className='Catalogs'>
			<h2>Catalogs</h2>
			<ul>
				<Fade when={menuIsActive} delay={200} duration={400} bottom>
					{allCatalogs}
				</Fade>
			</ul>
		</section>
	);
};

export default Catalogs;
