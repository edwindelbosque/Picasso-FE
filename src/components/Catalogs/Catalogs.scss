.Catalogs {
	padding-left: 50px;
	overflow: scroll;
	direction: rtl;
	padding-top: 35px;
	width: 700px;
	height: 102%;
	padding-right: 30px;

	a {
		text-decoration: none;
		color: black;
	}

	li {
		font-size: 80px;
		list-style-type: none;
		cursor: pointer;
		transition: 0.15s;

		&:hover {
			color: white;
			text-decoration: underline;
			transform: translate(30px);
		}
	}
}
