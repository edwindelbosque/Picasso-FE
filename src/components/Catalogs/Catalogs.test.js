import React from 'react';
import Catalogs from './Catalogs';
import { shallow } from 'enzyme';

describe('Catalogs', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<Catalogs />);
	});

	it('App should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
