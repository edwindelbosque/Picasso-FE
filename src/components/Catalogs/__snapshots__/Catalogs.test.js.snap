// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalogs App should match snapshot 1`] = `
<section
  className="Catalogs"
>
  <h2>
    Catalogs
  </h2>
  <ul>
    <Fade
      bottom={true}
      delay={200}
      duration={400}
    >
      <NavLink
        activeClassName="active-catalog"
        exact={true}
        key="1"
        to="/catalogs/1"
      >
         
        <li>
          Personal
        </li>
      </NavLink>
      <NavLink
        activeClassName="active-catalog"
        exact={true}
        key="2"
        to="/catalogs/2"
      >
         
        <li>
          Favorites
        </li>
      </NavLink>
      <NavLink
        activeClassName="active-catalog"
        exact={true}
        key="3"
        to="/catalogs/3"
      >
         
        <li>
          Mood
        </li>
      </NavLink>
      <NavLink
        activeClassName="active-catalog"
        exact={true}
        key="4"
        to="/catalogs/4"
      >
         
        <li>
          Work
        </li>
      </NavLink>
      <NavLink
        activeClassName="active-catalog"
        exact={true}
        key="5"
        to="/catalogs/5"
      >
         
        <li>
          Favorites
        </li>
      </NavLink>
      <NavLink
        activeClassName="active-catalog"
        exact={true}
        key="6"
        to="/catalogs/6"
      >
         
        <li>
          Mood
        </li>
      </NavLink>
      <NavLink
        activeClassName="active-catalog"
        exact={true}
        key="7"
        to="/catalogs/7"
      >
         
        <li>
          Work
        </li>
      </NavLink>
      <NavLink
        activeClassName="active-catalog"
        exact={true}
        key="8"
        to="/catalogs/8"
      >
         
        <li>
          Favorites
        </li>
      </NavLink>
      <NavLink
        activeClassName="active-catalog"
        exact={true}
        key="9"
        to="/catalogs/9"
      >
         
        <li>
          Mood
        </li>
      </NavLink>
      <NavLink
        activeClassName="active-catalog"
        exact={true}
        key="10"
        to="/catalogs/10"
      >
         
        <li>
          Work
        </li>
      </NavLink>
    </Fade>
  </ul>
</section>
`;
